packages/tempus/src/Tempus_StepperBDF2_impl.hpp

```diff
@@ -131,20 +131,17 @@ template<class Scalar>
 void StepperBDF2<Scalar>::setObserver(
   Teuchos::RCP<StepperObserver<Scalar> > obs)
 {
-  if (obs == Teuchos::null) {
-    // Create default observer, otherwise keep current observer.
-    if (this->stepperObserver_ == Teuchos::null) {
-      stepperBDF2Observer_ =
-        Teuchos::rcp(new StepperBDF2Observer<Scalar>());
-      this->stepperObserver_ =
-        Teuchos::rcp_dynamic_cast<StepperObserver<Scalar> >
-          (stepperBDF2Observer_);
-     }
-  } else {
-    this->stepperObserver_ = obs;
-    stepperBDF2Observer_ =
-      Teuchos::rcp_dynamic_cast<StepperBDF2Observer<Scalar> >(this->stepperObserver_);
-  }
+
+  if (this->stepperObserver_ == Teuchos::null)
+    this->stepperObserver_  =
+      Teuchos::rcp(new StepperObserverComposite<Scalar>());
+
+  if (obs == Teuchos::null)
+    obs = Teuchos::rcp(new StepperBDF2Observer<Scalar>());
+
+  this->stepperObserver_->addObserver(
+      Teuchos::rcp_dynamic_cast<StepperObserver<Scalar> > (obs, true) );
+
```

Comment:
So the new setObservers will not work the Stepper's initialize(), as initialize() can be called multiple times and will call setObserver without an argument and would add the default observer multiple times.

Reply:
So I was thinking about inspecting the size of the composite observer but that would require either changing the vector to a public or adding an a method. What do you think?

Reply:
Leaning towards encapsulation, I would go with adding a method.

Reply:
Line 139 probably needs the conditional (this->stepperObserver_->getSize() == 0)